### DISCUSSION TITLE
Macro to advance the calendar one day

### DISCUSSION THREAD

**Discussion by Hoosjan** [vigoren/foundryvtt-simple-calendar#546] [General]
Hi all! 

I'd love a macro that I can click to advance the day by one. Sometimes I forget to advance the calendar, by having that macro there I can remind myself and make it easier to advance it. 

Has anyone made one already?

**Comment by cwlithgow** [comment]
I too would be interested in this. I want to combine it with another (healing) macro so with one click I can run both

**Reply by Hoosjan** [reply]
I remembered I wanted this again after running in to the clicks required during sessions. I looked through the documentation and found some code I could use. With a little help I ended up making this: 
```

(async () => {
    // Stap 1: Huidige dag ophalen
    let currentDay = SimpleCalendar.api.getCurrentDay();
    let yesterday = currentDay.name; // Stap 2: Opslaan als "gister" variabele

    // Stap 3: Dag 1 verder verschuiven
    await SimpleCalendar.api.changeDate({day: +1});

    // Stap 4: Huidige dag opnieuw ophalen
    currentDay = SimpleCalendar.api.getCurrentDay();
    let today = currentDay.name; // Stap 4: Opslaan als "Huidige dag" variabele

    // Stap 5: Melding tonen met de verschoven dagen
    ui.notifications.info(`De dag is verplaatst van ${yesterday} naar ${today}.`);
})();
```

**Reply by Hoosjan** [reply]
@cwlithgow I also have a macro to add a note to current day if you're interested.
